### PR TITLE
Rename Pre-Main class

### LINKED ISSUE #924: Rename io.opentelemetry.auto.bootstrap.AgentBootstrap
Updated javaagent-jar-components.md in #884 makes it seem strange that this class named "bootstrap" is the 1 class that is not in the bootstrap class loader. It does "bootstrap" the agent, but I think better to rename and avoid this confusion. Since this is the `Premain-Class`, we should probably do this on the sooner side if we are going to do it.

As far as what to rename it to.... maybe `io.opentelemetry.auto.Agent`?

### PR DESCRIPTION
**Pull request by trask**
Closes #924

### CODE REVIEW

File: opentelemetry-javaagent/src/main/java/io/opentelemetry/javaagent/OpenTelemetryAgent.java [hunk 1]
```diff
@@ -14,7 +14,7 @@
  * limitations under the License.
  */
 
-package io.opentelemetry.auto.bootstrap;
+package io.opentelemetry.javaagent;
```
Comment (iNikem) [inline opentelemetry-javaagent/src/main/java/io/opentelemetry/javaagent/OpenTelemetryAgent.java:17]: Are you sure we don't want to keep `auto` part? `io.opentelemetry.auto.javaagent`?

File: agent-bootstrap/src/main/java/io/opentelemetry/auto/bootstrap/Agent.java [hunk 2]
```diff
@@ -27,7 +27,7 @@
 /**
```
Comment (anuraaga) [inline agent-bootstrap/src/main/java/io/opentelemetry/auto/bootstrap/Agent.java:27]: We probably need a better name for this file, `AgentBootstrap`, `AgentDelegate`, etc

File: opentelemetry-javaagent/src/main/java/io/opentelemetry/javaagent/OpenTelemetryAgent.java (hunk 1 above)
Comment (trask) [reply opentelemetry-javaagent/src/main/java/io/opentelemetry/javaagent/OpenTelemetryAgent.java:17]: I did this to be forward compatible with the proposal in https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/754#issuecomment-671011320, in particular this part at the end of that comment:

* io.opentelemetry.javaagent.tooling
* io.opentelemetry.javaagent.bootstrap

File: agent-bootstrap/src/main/java/io/opentelemetry/auto/bootstrap/Agent.java (hunk 2 above)
Comment (trask) [reply agent-bootstrap/src/main/java/io/opentelemetry/auto/bootstrap/Agent.java:27]: 👍 will do this as part of https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/754#issuecomment-673275367

**Pull request merged**